README.md

```diff
@@ -101,47 +101,90 @@ libp2p becomes very simple and basically acts as a glue for every module that co
 ```JavaScript
 // Creating a bundle that adds:
 //   transport: websockets + tcp
-//   stream-muxing: SPDY
+//   stream-muxing: spdy & mplex
 //   crypto-channel: secio
 //   discovery: multicast-dns
 
 const libp2p = require('libp2p')
 const TCP = require('libp2p-tcp')
 const WS = require('libp2p-websockets')
 const spdy = require('libp2p-spdy')
+const mplex = require('libp2p-mplex')
 const secio = require('libp2p-secio')
 const MulticastDNS = require('libp2p-mdns')
 const DHT = require('libp2p-kad-dht')
+const defaultsDeep = require('lodash.defaultsdeep')
 
 class Node extends libp2p {
-  constructor (peerInfo, peerBook, options) {
-    options = options || {}
-
-    const modules = {
-      transport: [
-        new TCP(),
-        new WS()
-      ],
-      connection: {
-        muxer: [
-          spdy
+  constructor (_peerInfo, _peerBook, _options) {
+    const defaults = {
+      peerInfo: _peerInfo             // The Identity of your Peer
+      peerBook: _peerBook,            // Where peers get tracked, if undefined libp2p will create one instance
+
+      // The libp2p modules for this libp2p bundle
+      modules: {
+        transport: [
+          TCP,
+          new WS()                    // It can take instances too!
         ],
-        crypto: [
-          secio
+        streamMuxer: [
+          SPDY
+        ],
+        connEncryption: [
+          SECIO
         ]
+        peerDiscovery: [
+          MulticastDNS
+        ],
+        peerRouting: {},              // Currently both peerRouting and contentRouting are patched through the DHT,
+        contentRouting: {}            // this will change once we factor that into two modules, for now do the following line:
+        dht: DHT                      // DHT enables PeerRouting, ContentRouting and DHT itself components
+      },
+
+      // libp2p config options (typically found on a config.json)
+      config: {                       // The config object is the part of the config that can go into a file, config.json.
+        addresses: {                  // Multiaddrs
+          listen: []                  // To start transport listener
+          announce: []                // To announce/share with the network (i.e DNS addr)
+          noAnnounce: []              // To keep private
+        },
```

Comment:
These are extra address to the PeerInfo.

- [ ] Impl this feature
- [ ] Add tests
  - [ ] listen
  - [ ] announce
  - [ ] noAnnounce

Reply:
Will break this into a separate issue to not delay this refactor any longer

Reply:
https://github.com/libp2p/js-libp2p/issues/202